app/controllers/posts_controller.rb

```diff
@@ -6,4 +6,65 @@ def index
   def show
     @post = Post.where(user_id: params[:user_id]).find(params[:id])
   end
+
+  def new
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def new_comment
+    comment = Comment.new
+    respond_to do |format|
+      format.html { render :new_comment, locals: { comment: } }
+    end
+  end
+
+  def create_comment
+    comment = Comment.new(params.require(:comment).permit(:text))
+    comment.user = current_user
+    comment.post = Post.find(params[:id])
+
+    respond_to do |format|
+      format.html do
+        if comment.save
+          flash[:success] = 'Comment created successfully'
+          redirect_to '/users/1/posts/1/'
+        else
```

Comment:
- **[OPTIONAL]** Kindly remember to update the post's comments count after a comment has been added.

  This can be done by simply changing modifying your Comment model to this:
  ```
  ...
  after_save :update_comments_counter

  def update_comments_counter
    post.increment!(:CommentsCounter)
  end
  ```

Reply:
a much better way of incrementing the counters! I agree that this behavior should be contained in the model. So when it's created, the counter is immediately updated.